python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -682,3 +685,141 @@ def _makeArgumentParser(cls):
                                "e.g. --id visit=12345 ccd=1,2 [tract=0]",
                                ContainerClass=PerTractCcdDataIdContainer)
         return parser
+
+
+class ForcedPhotCcdFromParquetConnections(ForcedPhotCcdConnections,
+                                          dimensions=("instrument", "visit", "detector", "skymap", "tract"),
+                                          defaultTemplates={"inputCoaddName": "goodSeeing",
+                                                            "inputName": "calexp"}):
+    refCat = cT.Input(
+        doc="Catalog of positions at which to force photometry.",
+        name="{inputCoaddName}Diff_fullDiaObjTable",
+        storageClass="DataFrame",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+        deferLoad=True,
+    )
+    measCat = cT.Output(
+        doc="Output forced photometry catalog.",
+        name="forced_src_diaObject",
+        storageClass="SourceCatalog",
+        dimensions=["instrument", "visit", "detector", "skymap", "tract"],
+    )
+    outputSchema = cT.InitOutput(
+        doc="Schema for the output forced measurement catalogs.",
+        name="forced_src_diaObject_schema",
+        storageClass="SourceCatalog",
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        self.initInputs.remove("inputSchema")
+        self.inputs.remove("skyMap")
```

Comment:
This is a sign that it'd be better to make a new ABC for both concrete implementations, instead of inheriting from the other concrete impl.  I'd say that's a job for another ticket, but these are going to still show up in the `config.connections` for this task, and it can be a real pain to remove configs with breakage.  Some copy-and-paste duplication would probably be better.

Reply:
Thanks, what I'm reading is that:
1) First choice is an ABC 
2) Second choice is a non-inherited Connections class.

Reply:
Yup, exactly.

Reply:
...and I'm not super confident that (1) is even better than (2); I'm just guessing that is because you chose to inherit in the first place.